Src/library/net/network/socket/network_socket.e

```diff
@@ -116,76 +116,92 @@ feature -- Status report
 			-- Is reuse_address option set?
 		require
 			socket_exists: exists
-		local
-			reuse: INTEGER
 		do
-			reuse := c_get_sock_opt_int (descriptor, level_sol_socket, so_reuse_addr);
-			Result := reuse /= 0
+			Result := c_get_sock_opt_int (descriptor, level_sol_socket, so_reuse_addr) /= 0
```

Comment:
I think the goal here was to use a local to see what the error code was.

Reply:
I agree as well, I changed to follow Alexander's suggestion.
Now it would even be better to store that value somewhere else, so that caller could check it too.
But this would be for another patch.